### PR TITLE
Grid Layout and Applying Oat Background

### LINKED ISSUE #417: Set background colour of the article body content
**Is your feature request related to a problem? Please describe.**
The background colour of the BodyContent is white

**Describe the solution you'd like**
The background colour should be `#F2EFEC` aka $OAT

**Describe alternatives you've considered**
N/A

**Additional context**
main content background colour

**Testing notes**
- Snapshot tests would cover this. Cypress tests are not necessary.

### PR DESCRIPTION
**Pull request by sareh**
Resolves #417

* This PR updates two sets of styles `layoutGridWrapper` and `layoutGridItemConstrained`. The wrapper is used at the top-most level. Once to wrap the heading the headline and timestamp components, and the once to wrap the remainder of the article body content. The latter wrapper has an Oat Light coloured background. 
* For our alpha deadline we do not have a requirement for components to span outside of the grid and therefore we only require a single grid item within each grid. 
* Corrects values of the constants `group2ScreenWidthMin` and `group2ScreenWidthMax`
* Adds padding to the header component above 1007px to stop the PR adding a visual regression. The need to fix this arose from fixing the constant values above. 
* In this PR, we have explicitly *not* removed the grid layout from the Brand Header - this will be done in a separate PR. 

- [x] Snapshots updated
- [ ] ~Tests added for new features~
- [ ] Test engineer approval

### CODE REVIEW

File: src/app/lib/constants/styles.js [hunk 1]
```diff
@@ -59,7 +59,7 @@ export const group0ScreenWidthMax = `14.9375em`; // 239px
 export const group1ScreenWidthMin = `15em`; // 240px
 export const group1ScreenWidthMax = `24.9375em`; // 399px
 
-export const group2ScreenWidthMin = `37.5em`; // 400px
+export const group2ScreenWidthMin = `25em`; // 400px
```
Comment (pjlee11) [inline src/app/lib/constants/styles.js:72]: `37.5 * 16 !== 400px;`

**Comment by pjlee11** [comment]
### 399px and below
380px

### 400px up to 599px
420px

### 600px up to 1007px
620px

### 1008px up to 1279px
1008px

### 1280px and above
1280px

### 1440px
1440px

**Comment by pjlee11** [comment]
### UX margins/columns
* UX margins are exclusive of the UX columns. 
EG: At 500px there are margins of 16px either side of the column definitions, the columns then have 8px gutters. 

image

* UX columns above 1008px have statically defined widths. 
EG: At 1600px the UX grid is restricted to 1280px and the remaining viewport outside the 1280px is flexible. 

image

### Our implementation
* Above 1007px the UX grid is limited in width and our implementation creates the columns as static widths allowing us to create two additional columns of flexible width either side. Therefore above 1279px we have 12 columns and above 1007px we have 10 columns. Because of these additional columns we have additional gutters which act as the UX margin's for layouts above 1007px. 

* Below the 1008px the UX grid is responsive the to full width of the viewport and therefore we do not require the additional two columns. Because of this we implement the UX margins as padding on the grid itself.

File: src/app/components/Brand/index.jsx [hunk 2]
```diff
@@ -31,6 +33,9 @@ const StyledGridWrapper = styled.div`
   background-color: ${C_POSTBOX};
   height: ${BANNER_HEIGHT};
   width: 100%;
+  @media (min-width: ${group3ScreenWidthMax}) {
+    padding: 0 ${GEL_SPACING_DBL};
```
Comment (pjlee11) [inline src/app/components/Brand/index.jsx:37]: Due to the implementation detail described in the [comments of this PR](https://github.com/BBC-News/simorgh/pull/874#issuecomment-434427025), the Brand component suffers a visual regression above 1007px due to it previously relying on padding on the grid wrapper. As the padding has been removed above 1007px (see implementation detail for explanation) we need to add the padding on the Brand component. 

This is a temporary solution until the [grid is completely removed from the branch component](https://github.com/BBC-News/simorgh/issues/827). 

NB: If others feel like the regression is not enough for this change to be included in this work I'm happy to open a bug ticket and fix forward in a follow-up PR.

File: src/app/lib/constants/styles.js [hunk 3]
```diff
@@ -59,8 +59,8 @@ export const group0ScreenWidthMax = `14.9375em`; // 239px
 export const group1ScreenWidthMin = `15em`; // 240px
 export const group1ScreenWidthMax = `24.9375em`; // 399px
 
-export const group2ScreenWidthMin = `37.5em`; // 400px
-export const group2ScreenWidthMax = `56.1875em`; // 599px
+export const group2ScreenWidthMin = `25em`; // 400px
+export const group2ScreenWidthMax = `37.4375`; // 599px
```
Comment (pjlee11) [inline src/app/lib/constants/styles.js]: `56.1875 * 16 !== 599px`

File: src/app/lib/layoutGrid.js [hunk 4]
```diff
@@ -2,47 +2,56 @@ import { css } from 'styled-components';
...
+/* (1280px width - 9 * 16px gap) / 10 columns = 113.6px = 7.1rem single column width */
+
 export const layoutGridWrapper = css`
   display: grid;
-  @media (max-width: ${group3ScreenWidthMin}) {
-    grid-gap: ${GEL_SPACING}; /* Between grid items */
-    padding: 0 ${GEL_SPACING}; /* On grid wrapper */
+  @media (max-width: ${group2ScreenWidthMin}) {
+    grid-gap: ${GEL_SPACING};
+    padding: 0 ${GEL_SPACING};
```
Comment (pjlee11) [inline src/app/lib/layoutGrid.js]:
```suggestion
    padding: 0 ${GEL_SPACING}; // UX margin exclusive of grid
```

File: src/app/lib/layoutGrid.js [hunk 5]
```diff
@@ -2,47 +2,56 @@ import { css } from 'styled-components';
...
+  }
+  @media (min-width: ${group2ScreenWidthMin}) and (max-width: ${group3ScreenWidthMax}) {
+    padding: 0 ${GEL_SPACING_DBL};
```
Comment (pjlee11) [inline src/app/lib/layoutGrid.js]:
```suggestion
    padding: 0 ${GEL_SPACING_DBL}; // UX margin exclusive of grid
```

File: src/app/lib/layoutGrid.js [hunk 6]
```diff
@@ -2,47 +2,56 @@ import { css } from 'styled-components';
...
+const group5Const = `7.1rem`;
+/* (1280px width - 9 * 16px gap) / 10 columns = 113.6px = 7.1rem single column width */
+
 export const layoutGridWrapper = css`
   display: grid;
-  @media (max-width: ${group3ScreenWidthMin}) {
-    grid-gap: ${GEL_SPACING}; /* Between grid items */
-    padding: 0 ${GEL_SPACING}; /* On grid wrapper */
+  @media (max-width: ${group2ScreenWidthMin}) {
+    grid-gap: ${GEL_SPACING};
```
Comment (pjlee11) [inline src/app/lib/layoutGrid.js]:
```suggestion
    grid-gap: ${GEL_SPACING}; // UX gutter
```

File: src/app/lib/layoutGrid.js [hunk 7]
```diff
@@ -2,47 +2,56 @@ import { css } from 'styled-components';
...
+  @media (min-width: ${group2ScreenWidthMin}) and (max-width: ${group3ScreenWidthMax}) {
+    padding: 0 ${GEL_SPACING_DBL};
   }
   @media (min-width: ${group3ScreenWidthMin}) {
-    grid-gap: ${GEL_SPACING_DBL}; /* Between grid items */
-    padding: 0 ${GEL_SPACING_DBL}; /* On grid wrapper */
+    grid-gap: ${GEL_SPACING_DBL};
   }
   @media (max-width: ${group3ScreenWidthMax}) {
+    grid-gap: ${GEL_SPACING_DBL};
```
Comment (pjlee11) [inline src/app/lib/layoutGrid.js]:
```suggestion

```

Comment (pjlee11) [reply src/app/lib/layoutGrid.js]: Covered by line 29

File: src/app/lib/layoutGrid.js [hunk 8]
```diff
@@ -2,47 +2,56 @@ import { css } from 'styled-components';
...
   }
   @media (min-width: ${group5ScreenWidthMin}) {
-    grid-template-columns: repeat(10, 1fr);
+    grid-template-columns: 1fr repeat(10, minmax(0, ${group5Const})) 1fr;
   }
 `;
 
 export const layoutGridItem = css`
-  @media (max-width: ${group2ScreenWidthMax}) {
+  @media (max-width: ${group3ScreenWidthMin}) {
```
Comment (pjlee11) [inline src/app/lib/layoutGrid.js:57]: This changed as the constant value was incorrect for `group2ScreenWidthMax`

File: src/app/lib/layoutGrid.js [hunk 9]
```diff
@@ -2,47 +2,56 @@ import { css } from 'styled-components';
...
+  group2ScreenWidthMin,
   group3ScreenWidthMin,
   group3ScreenWidthMax,
   group4ScreenWidthMin,
   group4ScreenWidthMax,
   group5ScreenWidthMin,
 } from './constants/styles';
 
+const group4Const = `7rem`;
+/* (1008px width - 7 * 16px gap) / 8 columns = 112px = 7rem single column width */
```
Comment (pjlee11) [inline src/app/lib/layoutGrid.js]:
```suggestion
/* (1008px width - ( 2 * 16px UX margin + 5 * 16px gutter)) / 8 columns = 112px = 7rem single column width */
```

File: src/app/lib/layoutGrid.js [hunk 10]
```diff
@@ -2,47 +2,56 @@ import { css } from 'styled-components';
...
+
+const group5Const = `7.1rem`;
+/* (1280px width - 9 * 16px gap) / 10 columns = 113.6px = 7.1rem single column width */
```
Comment (pjlee11) [inline src/app/lib/layoutGrid.js]:
```suggestion
/* (1280px width - (2 * 16px Ux margin + 7 * 16px gutter) / 10 columns = 113.6px = 7.1rem single column width */
```

**Comment by sareh** [comment]
The CodeClimate duplication warning appears to be overzealous. (It notes that there is a `display: grid` line in two different files.) I suggest ignoring the warning.

File: src/app/components/MainContent/index.jsx [hunk 11]
```diff
@@ -1,38 +1,7 @@
...
-`;
-
-const MainContent = ({ children }) => (
-  <StyledMainContent role="main">
-    <StyleGridWrapper>
-      <GridItem>{children}</GridItem>
-    </StyleGridWrapper>
-  </StyledMainContent>
-);
+const MainContent = ({ children }) => <main role="main">{children}</main>;
```
Comment (jtart) [inline src/app/components/MainContent/index.jsx]: This component could be removed and just use a `main` element [here](https://github.com/BBC-News/simorgh/pull/874/files#diff-d5c925f82e0315a863ea6a24dd59cb60R80) now.

File: src/app/containers/Blocks/index.jsx [hunk 12]
```diff
@@ -8,14 +8,23 @@ const BlockString = props => {
...
-const Blocks = ({ blocks, componentsToRender }) =>
+const Blocks = ({ blocks, componentsToRender, isTopLevel }) =>
   blocks.map((block, index) => {
     const { type, model } = block;
 
     const { type: typeOfPreviousBlock } = blocks[index - 1] || {};
 
     const Block = componentsToRender[type] || BlockString;
-
+    if (isTopLevel) {
```
Comment (jtart) [inline src/app/containers/Blocks/index.jsx]: This logic and prop isn't doing anything different in this PR, can it be removed?

File: src/app/lib/layoutGrid.js [hunk 13]
```diff
@@ -1,48 +1,69 @@
...
+  GEL_GUTTER_ABOVE_600PX,
+  group2ScreenWidthMin,
   group3ScreenWidthMin,
   group3ScreenWidthMax,
   group4ScreenWidthMin,
   group4ScreenWidthMax,
   group5ScreenWidthMin,
 } from './constants/styles';
 
+const group4ColWidth = `6.75rem`;
```
Comment (jtart) [inline src/app/lib/layoutGrid.js:16]: Can these be moved to `constant/styles`?

File: src/app/lib/constants/styles.js [hunk 14]
```diff
@@ -38,6 +38,16 @@ export const GEL_SPACING_TRPL = `1.5rem`;
 export const GEL_SPACING_HLF = `0.25rem`;
 export const GEL_SPACING_QUAD = `2rem`;
 
+/*
+    GEL Grid
+    Margins and Gutters are defined here
+    https://www.bbc.co.uk/gel/guidelines/grid#grid-sizes
+*/
+export const GEL_MARGIN_BELOW_400PX = `${GEL_SPACING}`;
```
Comment (jtart) [inline src/app/lib/constants/styles.js:46]: Do these need to be variables, can we just use `GEL_SPACING` or `GEL_SPACING_DBL` directly?

File: src/app/containers/Article/index.jsx [hunk 15]
```diff
@@ -11,6 +12,20 @@ import MainContent from '../../components/MainContent';
...
+
+const Wrapper = styled.div`
+  ${layoutGridWrapper};
+`;
+
+const OatWrapper = styled(Wrapper)`
+  background: ${C_OAT_LHT};
+`;
+
+const Item = styled.div`
```
Comment (ChrisBAshton) [inline src/app/containers/Article/index.jsx]: To me this seems a bit of a vague naming - maybe better as `GridItem` or `ConstrainedItem`?

File: src/app/containers/Blocks/index.jsx (hunk 12 above)
Comment (ChrisBAshton) [reply src/app/containers/Blocks/index.jsx]: Agreed, let's not abstract too early!

File: src/app/lib/constants/styles.js (hunk 14 above)
Comment (sareh) [reply src/app/lib/constants/styles.js:46]: @jtart @ChrisBAshton 
The GEL SPACING variables were initially used directly. These `GEL_MARGIN_*` `GEL_GUTTER_*` values however *are* constants that GEL Grid has given us. (https://www.bbc.co.uk/gel/guidelines/grid#grid-sizes)
It makes the code in the `layoutGrid.js` file easier to read and removes the need for suggested comments like `// UX gutter` `// UX margin exclusive of grid` as explanations. It is self-documenting.

File: src/app/containers/Article/index.jsx (hunk 15 above)
Comment (sareh) [reply src/app/containers/Article/index.jsx]: Agreed. `GridItemConstrained` for this one, so we can later have a `GridItem` that is full-width. I will also change the variable in the layoutGrid file to be called `layoutGridItemConstrained`, for consistency.

File: src/app/lib/constants/styles.js (hunk 14 above)
Comment (ChrisBAshton) [reply src/app/lib/constants/styles.js:46]: I like self-documenting 👍

File: src/app/lib/layoutGrid.js (hunk 13 above)
Comment (pjlee11) [reply src/app/lib/layoutGrid.js:16]: These are very much bespoke to the 1280px/1008px columns rendering logic (due to the margins being exclusive of the grid and the gutters being inclusive. The actual values are `group4ColWidth == 108px` and `group5ColWidth == 110.4` - it is highly unlikely another file will re-use them, I'd even want to question why they would need to be used in the future so would suggest keeping them here.

File: src/app/containers/Blocks/index.jsx (hunk 12 above)
Comment (pjlee11) [reply src/app/containers/Blocks/index.jsx]: Good spot, now that each grid only contains a single wrapping item, which is created in the article container, this logic can be removed.

**Review by ChrisBAshton — DISMISSED** [review]

File: src/app/containers/Blocks/index.jsx [hunk 16]
```diff
@@ -34,6 +34,11 @@ Blocks.propTypes = {
     }),
   ).isRequired,
   componentsToRender: objectOf(func).isRequired,
+  isTopLevel: bool,
```
Comment (jtart) [inline src/app/containers/Blocks/index.jsx]: These props don't serve any use, can they be removed?

File: src/app/containers/Blocks/index.test.jsx [hunk 17]
```diff
@@ -54,4 +54,13 @@ describe('Blocks', () => {
     'should render correctly',
     <Blocks blocks={blocks} componentsToRender={componentsToRender} />,
   );
+
+  shouldShallowMatchSnapshot(
```
Comment (jtart) [inline src/app/containers/Blocks/index.test.jsx]: This test can be removed based on this comment https://github.com/BBC-News/simorgh/pull/874/files#r229646193

**Review by ChrisBAshton — APPROVED** [review]
🎉

**Review by jtart — APPROVED** [review]

**Review by BogdanDogaru — APPROVED** [review]
👍 Nice work, team!

**Comment by pjlee11** [comment]
UX review 👍

**Pull request merged**